.github/workflows/main_workflow.yml

```diff
@@ -0,0 +1,87 @@
+name: Debianization
+on:
+  workflow_dispatch:
+    inputs:
+      repository:
+        description: "Aptly server repository"
+        required: true
+        default: "experimental"
+        type: choice
+        options:
+          - stable
+          - rc
+          - experimental
+          - test
+  push:
+concurrency:
+  group: ${{ github.ref }}-debianization
+  cancel-in-progress: true
+defaults:
+  run:
+    shell: bash
+env:
+  SERVER_HOSTNAME: aptly.greenzie.com
+jobs:
+  debianize:
+    name: "Debianization"
+    timeout-minutes: 30
+    runs-on: ubuntu-latest
+    container:
+      image: greenzie/debianize:focal-noetic
+      credentials:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_TOKEN }}
+    steps:
+      - name: Checkout Codebase
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.ref }}
+          submodules: false
+      - name: "Changelogs"
+        run: greenzie-release changelog -r "${ROS_DISTRO}"
+      - name: "Debianizing"
+        run: greenzie-debianize all
+      - name: Archive Artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: alldebs-${{ github.run_id }}
+          path: /debians/
+          retention-days: 1
+  deploy:
+    if: ${{ github.ref_name == 'noetic' || github.ref_name == 'noetic-devel' }}
```

Comment:
@AGummyBear   as you can see  we only deploy IF the branch is either `noetic` or `noetic-devel`